### PR TITLE
Added missing events functional tests with mock irqs

### PR DESCRIPTION
**Pull request by jelonanX**

### CODE REVIEW

File: testing/xfpga/test_events_c.cpp [hunk 1]
```diff
@@ -1021,12 +1109,12 @@ class events_handle_p : public ::testing::TestWithParam<std::string> {
...
 
-
-
 /**
  * @test       irq_event_01
  *
- * @brief
+ * @brief      given valid event handle and event type, this tests for
+ *             triggering the FPGA_EVENT_POWER_THERMAL event by mannually
+ *             writing Ap6Event to erriors.
```
Comment (tswhison) [inline testing/xfpga/test_events_c.cpp]: s/erriors/errors/

File: testing/xfpga/test_events_c.cpp [hunk 2]
```diff
@@ -1066,8 +1154,295 @@ TEST_P(events_handle_p, irq_event_01) {
...
+      : tokens_accel_{{nullptr, nullptr}},
+        tokens_dev_{{nullptr, nullptr}},
+        tmpfpgad_log_("tmpfpgad-XXXXXX.log"),
+        tmpfpgad_pid_("tmpfpgad-XXXXXX.pid"),
+        handle_accel_(nullptr),
+        handle_dev_(nullptr) {}
+
+  virtual void SetUp() override {
+    tmpfpgad_log_ = mkstemp(const_cast<char *>(tmpfpgad_log_.c_str()));
+    tmpfpgad_pid_ = mkstemp(const_cast<char *>(tmpfpgad_pid_.c_str()));
```
Comment (tswhison) [inline testing/xfpga/test_events_c.cpp]: These two lines are incorrect.  mkstemp actually returns an open file descriptor (int).  See testing/fpgad/test_evt_c.cpp for an example.  I think we still have some instances of this to scrub from other tests as well.

File: testing/xfpga/test_events_c.cpp [hunk 3]
```diff
@@ -1066,8 +1154,295 @@ TEST_P(events_handle_p, irq_event_01) {
   ASSERT_EQ(FPGA_OK, sysfs_write_u64(path.c_str(), error_csr));
   EXPECT_EQ(FPGA_OK, xfpga_fpgaUnregisterEvent(handle_accel_, FPGA_EVENT_POWER_THERMAL, eh_));
 }
-
-//INSTANTIATE_TEST_CASE_P(events, events_handle_p,
-//                        ::testing::ValuesIn(test_platform::keys()));
 INSTANTIATE_TEST_CASE_P(events, events_handle_p,
                         ::testing::Values("skx-p-1s"));
+
+class mock_events_handle_p : public testing::TestWithParam<std::string> {
```
Comment (ibarraza) [inline testing/xfpga/test_events_c.cpp]: These two classes are almost identical, the only difference being this one also opens the fme device. Can we merge these classes?

File: testing/xfpga/test_events_c.cpp (hunk 2 above)
Comment (jelonanX) [reply testing/xfpga/test_events_c.cpp]: Will fix those.

File: testing/xfpga/test_events_c.cpp [hunk 4]
```diff
@@ -182,19 +220,63 @@ int set_irq(mock_object * m, int request, va_list argp){
...
         handle_dev_(nullptr),
         handle_accel_(nullptr) {}
 
   virtual void SetUp() override {
-    tmpfpgad_log_ = mkstemp(const_cast<char *>(tmpfpgad_log_.c_str()));
-    tmpfpgad_pid_ = mkstemp(const_cast<char *>(tmpfpgad_pid_.c_str()));
+    strcpy(tmpfpgad_log_, "tmpfpgad-XXXXXX.log");
+    tmpfpgad_log_fd_ = mkstemps(tmpfpgad_log_, 4);
+    strcpy(tmpfpgad_pid_, "tmpfpgad-XXXXXX.pid");
+    tmpfpgad_pid_fd_ = mkstemps(tmpfpgad_pid_, 4);
```
Comment (tswhison) [inline testing/xfpga/test_events_c.cpp]: Let's not track the fds.  I convinced myself it's not necessary.  Instead..
  close(mkstemps(tmpfpgad_log_, 4));
  close(mkstemps(tmpfpgad_pid_, 4));

File: testing/xfpga/test_events_c.cpp [hunk 5]
```diff
@@ -259,13 +341,18 @@ class events_p : public ::testing::TestWithParam<std::string> {
     if (handle_dev_) { EXPECT_EQ(xfpga_fpgaClose(handle_dev_), FPGA_OK); }
     if (handle_accel_) { EXPECT_EQ(xfpga_fpgaClose(handle_accel_), FPGA_OK); }
     fpgad_.join();
+    close_log();
+    close(tmpfpgad_log_fd_);
+    close(tmpfpgad_pid_fd_);
     system_->finalize();
```
Comment (tswhison) [inline testing/xfpga/test_events_c.cpp]: Please add the code to unlink the temp files when there are no test failures.

File: testing/xfpga/test_events_c.cpp [hunk 6]
```diff
@@ -938,22 +1042,209 @@ TEST_P(events_p, create_destroy_invalid) {
...
-        tmpfpgad_pid_("tmpfpgad-XXXXXX.pid"),
         handle_accel_(nullptr) {}
 
   virtual void SetUp() override {
-    tmpfpgad_log_ = mkstemp(const_cast<char *>(tmpfpgad_log_.c_str()));
-    tmpfpgad_pid_ = mkstemp(const_cast<char *>(tmpfpgad_pid_.c_str()));
+    strcpy(tmpfpgad_log_, "tmpfpgad-XXXXXX.log");
+    tmpfpgad_log_fd_ = mkstemps(tmpfpgad_log_, 4);
+    strcpy(tmpfpgad_pid_, "tmpfpgad-XXXXXX.pid");
+    tmpfpgad_pid_fd_ = mkstemps(tmpfpgad_pid_, 4);
```
Comment (tswhison) [inline testing/xfpga/test_events_c.cpp]: Let's not track the fd's after all.

File: testing/xfpga/test_events_c.cpp [hunk 7]
```diff
@@ -1002,14 +1296,17 @@ class events_handle_p : public ::testing::TestWithParam<std::string> {
     }
 
     if (handle_accel_) { EXPECT_EQ(xfpga_fpgaClose(handle_accel_), FPGA_OK); }
-    logger_thread_.join();
-    fpgad_.join();
+    close_log();
+    close(tmpfpgad_log_fd_);
+    close(tmpfpgad_pid_fd_);
```
Comment (tswhison) [inline testing/xfpga/test_events_c.cpp]: Please add the code to remove the temp files if there is no test failure.

**Pull request closed without merging**